src/bindings/python/test_commands/sideflux.py

```diff
@@ -0,0 +1,194 @@
+from __future__ import print_function
+import re
+import os
+import sys
+import json
+import subprocess
+import multiprocessing as mp
+import contextlib
+import errno
+import pprint
+import shutil
+import tempfile
+import time
+import Queue
+import pycotap
+
+# pprint.pprint(os.environ)
+flux_exe = ''
+if os.environ.get('CHECK_BUILDDIR', None) is not None:
+    flux_exe = os.path.abspath(os.environ['CHECK_BUILDDIR'] + '/src/cmd/flux')
+else:
+    flux_exe = os.path.abspath(
+        os.path.dirname(os.path.abspath(__file__)) + '/../../../cmd/flux')
+
+
+@contextlib.contextmanager
+def get_tmpdir():
+    d = tempfile.mkdtemp()
+    try:
+        yield d
+    finally:
+        shutil.rmtree(d)
+
+
+def consume(stream):
+    while True:
+        l = stream.readline()
+        if not l:
+            break
+        sys.stdout.write(l)
+
+
+class SideFlux(object):
+    def __init__(self, size=1):
+        global flux_exe
+        self.size = size
+
+    def start(self):
+        flux_command = [flux_exe, '--verbose', 'start',
+                        '--size={}'.format(self.size), '-o',
+                        '--verbose,-L,stderr', 'bash']
+        # """bash -c 'echo READY ; while true ; do sleep 1; done' """]
+        # print ' '.join(flux_command)
+        FNULL = open(os.devnull, 'w+')
+
+        self.sub = subprocess.Popen(
+            flux_command,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            close_fds=True,
+            preexec_fn=os.setsid,
+            # Start a process session to clean up brokers
+        )
+        print('echo READY', file=self.sub.stdin)
+        self.env_items = {}
+        while True:
+            line = self.sub.stdout.readline()
+            # print line
+            if line != '':
+                m = re.match(r"\s+(FLUX_[^=]+)=(.*)", line.rstrip())
+                if m:
+                    if os.environ.get('SIDEFLUX_DEBUG', False):
+                        print("setting", m.group(1), "to", m.group(2))
+                    v = m.group(2)
+                    if re.search(r'/\.\./', v):
+                        v = os.path.abspath(v)
+                    self.env_items[m.group(1)] = v
+                m = re.match(r'lt-flux-broker: FLUX_TMPDIR: (.*)',
```

Comment:
I started using this script for more than testing, so I'm not using lt-flux-broker, I had the module search my PATH and find the installed flux command, which uses flux-broker.  This results in the regex not matching.  Would this be a suitable compromise? `(lt-)?flux-broker: FLUX_TMPDIR: (.*)`.  And then changing the 2 following calls to `m.group(1)` to `m.group(2)`?

PS - I'm making this a note as opposed to a commit/PR because I cannot get the test_runner.t test to pass even after a pull/clean/rebuild.  I think something in my environment is messed up.